objects/client.c

```diff
@@ -723,12 +723,58 @@
  * identifier remain the same. This allow to match a spawn event to an actual
  * client.
  *
+ * This is used to display a different mouse cursor when the application is
+ * loading and also to attach some properties to the newly created client (like
+ * a `tag` or `floating` state).
+ *
+ * Some applications, like `xterm`, don't support startup notification. While
+ * not perfect, the adding the following code to `rc.lua` will mitigate the
+ * issue. Please note that this code is Linux specific.
+ *
+ *    local blacklisted_snid = setmetatable({}, {__mode = "v" })
+ *
+ *    --- Make startup notification work for some clients like XTerm. This is ugly
+ *    -- but works often enough to be useful.
+ *    local function fix_startup_id(c)
+ *        -- Prevent "broken" sub processes created by `c` to inherit its SNID
+ *        if c.startup_id then
+ *            blacklisted_snid[c.startup_id] = blacklisted_snid[c.startup_id] or c
+ *            return
+ *        end
+ *
+ *        if not c.pid then return end
+ *
+ *        -- Read the process environment variables
+ *        local f = io.open("/proc/"..c.pid.."/environ", "rb")
+ *
+ *        -- It will only work on Linux, that's already 99% of the userbase.
+ *        if not f then return end
+ *
+ *        local snid = f:read("*all"):match("STARTUP_ID=(.*)\0")
```

Comment:
Correct me if I am wrong, but Lua patterns are greedy, so this actually gets you the value of `STARTUP_ID` plus a weird mess of all following environment variables. Sigh, such fun...

Reply:
Ah, okay, so this relies on the C code mis-handling strings and truncating them at embedded zero bytes in the call to `a_strdup()`. That's... surprising and could arguably be called a bug, but whatever. Weird magic...

Reply:
Ok, If you insist

done